File: Framework/Atom.Net.Browsing.WebDriver/Protocol/BridgeMessage.cs
using System.Text.Json;

namespace Atom.Net.Browsing.WebDriver.Protocol;

/// <summary>
/// Сообщение протокола обмена между драйвером и browser bridge.
/// </summary>
internal sealed class BridgeMessage
{
    public required string Id { get; init; }

    public required BridgeMessageType Type { get; init; }

    public string? WindowId { get; init; }

    public string? TabId { get; init; }

    public BridgeCommand? Command { get; init; }

    public BridgeEvent? Event { get; init; }

    public BridgeStatus? Status { get; init; }

    public JsonElement? Payload { get; init; }

    public string? Error { get; init; }

    public long Timestamp { get; init; } = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
}

File: Framework/Atom.Net.Browsing.WebDriver/ExtensionRuntime/Shared/Protocol/BridgeMessage.ts
import type { BridgeCommand } from './BridgeCommand';
import type { BridgeEventName } from './BridgeEventName';
import type { BridgeMessageType } from './BridgeMessageType';
import type { BridgeStatus } from './BridgeStatus';
import type { JsonValue } from './JsonValue';

export interface BridgeMessage {
    id: string;
    type: BridgeMessageType;
    windowId?: string;
    tabId?: string;
    command?: BridgeCommand;
    event?: BridgeEventName;
    status?: BridgeStatus;
    payload?: JsonValue;
    error?: string;
    timestamp?: number;
}